doc/api/cli.md

```diff
@@ -127,6 +127,16 @@ added: v12.0.0
 
 Specify the file name of the CPU profile generated by `--cpu-prof`.
 
+### `--dev`
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+Enables the `"development"` [conditional export][] in package resolution,
+while also setting `process.env.NODE_ENV` to `"development"`.
```

Comment:
would this override it if it's already set?

Reply:
Yes, the logic being that it could already exist in the outer environment, but that the inner environment source of truth is being informed by the flag.

Reply:
Is there any way to activate the condition without overriding the env var? I have use cases for that.

Reply:
I’m not sure we should be “silently” setting the env var here at all…

Reply:
An alternative could be to create a new env var for this entirely. Or another mechanism of checking the state internally.

That does mean having library authors write new patterns for:

```js
if (DEVELOPMENT) {
  // ... debugging
}
```

So opens up that can of worms instead of paving the existing cowpath.

Reply:
Their entry point could easily do `if (DEVELOPMENT) { process.env.NODE_ENV = 'development'; }`, and the node would continue to not interact with NODE_ENV at all, which I believe is the preference?

Reply:
Yes, if we are digging back into this difficult discussion then simply leaving the environment variable out for now until a viable pattern emerges does seem sensible. I'll update the PR.